### ISSUE TITLE
etl::queue_spsc_atomic::size under-count

### ISSUE THREAD

**Issue by John-Titor** [ETLCPP/etl#331]
This calculation:

`        n = RESERVED - read_index + write_index - 1;`

fails (at least) for a queue of size 128 when **read_index** is 128 and **write_index is 0**: `129 - 128 + 0 - 1 = 0`

I believe the -1 is incorrect; consider a queue of size 7:

```
read_index
           +
           |
+-----+-+--v----+
|0|1| | | |5|6|7|
+----^+-+-------+
     |
     +
     write_index
```
RESERVED = 8
read_index = 5
write_index = 2
8 - 5 + 2 - 1 = 4 <- but there are clearly 5 items in the queue.

**Comment by jwellbelove** [comment]
Thanks John, that looks suspicious!. I'll take a look

**Comment by jwellbelove** [comment]
Fixed 19.5.2
